gnovm/pkg/gnolang/preprocess.go

```diff
@@ -99,6 +108,9 @@ func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
 	}
 }
 
+// declGraph represents a slice of declGraph
+var declGraph []*DeclNode
```

Comment:
This should not be a global variable

It may be a good time to add a PreprocessContext, so we can also place the `preprocessing` global variable in there.

Reply:
> This should not be a global variable
> 
Yes, that’s right. However, I'm now exploring ways to potentially eliminate this new introduced dependency graph.

> It may be a good time to add a PreprocessContext, so we can also place the `preprocessing` global variable in there.

I agree.

Reply:
I am not in favor of increasing the scope of issues because X or Y could be better.
X or Y are separate issues that need to be fixed separately.
The existing PR solves the problem of the mentioned issue.

Reply:
AFAIK that is new code.

Reply:
implementing a new method to check.